### PR TITLE
[ISSUE-2498][Feature] [SubTask] The cluster supports remote and yarn session heartbeat monitoring

### PR DESCRIPTION
**Pull request by xujiangfeng001**
## What changes were proposed in this pull request

Issue Number: https://github.com/apache/incubator-streampark/issues/2498

This PR is a resubmission of https://github.com/apache/incubator-streampark/pull/2541. Compared to https://github.com/apache/incubator-streampark/pull/2541, this PR modifies the state change as follows:

Flink-Cluster状态

Cluster status:

- CREATED
- RUNNING
- STOPPED
- LOST
- UNKNOWN

If the state obtained from Yarn is UNDEFINED, it will be mapped to RUNNING of cluster status. If the states obtained from Yarn are FAILED, KILLED, SUCCEEDED and ENDED, they will be mapped as STOPPED for cluster status.

This pr supports status monitoring and status updates for the "Flink Cluster on Yarn" and "Remote".Heartbeat and status monitoring support for "Flink Cluster on K8s" will be supported in subsequent pr support.

## Brief change log

Nothing.

## Verifying this change

This change without any test coverage.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency):  no

### CODE REVIEW

**Comment by xujiangfeng001** [comment]
Hi @RocMarshal ,This is my resubmitted PR for https://github.com/apache/incubator-streampark/pull/2541. This PR has modified some suggestions for https://github.com/apache/incubator-streampark/pull/2541.

For the issue of https://github.com/apache/incubator-streampark/pull/2541#discussion_r1162910882, I have thought about it for a moment. In fact, it will not have an impact on obtaining the cluster status because the cluster failed to obtain the corresponding status information from` jobManagerUrl` and will go back to Yarn.

Looking forward to your next review.

**Comment by RocMarshal** [comment]
> Hi @RocMarshal ,This is my resubmitted PR for #2541. This PR has modified some suggestions for #2541.
> 
> For the issue of [#2541 (comment)](https://github.com/apache/incubator-streampark/pull/2541#discussion_r1162910882), I have thought about it for a moment. In fact, it will not have an impact on obtaining the cluster status because the cluster failed to obtain the corresponding status information from` jobManagerUrl` and will go back to Yarn.
> 
> Looking forward to your next review.

Thank you @xujiangfeng001  very much for your update. It seems closer to the ideal state.

Due to my personal reasons, the review may be conducted later.
Other reviewers CC @wolfboys @macksonmu  can also participate in this MR review.
If you have time on your end, you can try to continue to advance the design and discussion of other subtasks of https://github.com/apache/incubator-streampark/issues/2425
which will shorten some of the connection  time~
Best Roc.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java [hunk 1]
```diff
@@ -140,12 +141,16 @@ public Boolean create(FlinkCluster flinkCluster) {
     ApiAlertException.throwIfFalse(
         successful, String.format(ERROR_CLUSTER_QUEUE_HINT, flinkCluster.getYarnQueue()));
     flinkCluster.setCreateTime(new Date());
-    if (ExecutionMode.isRemoteMode(flinkCluster.getExecutionModeEnum())) {
-      flinkCluster.setClusterState(ClusterState.STARTED.getValue());
+    if (ExecutionMode.REMOTE.equals(flinkCluster.getExecutionModeEnum())) {
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java]: ExecutionMode.isRemote(...

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+  @Autowired private FlinkClusterService flinkClusterService;
+
+  private Long lastWatcheringTime = 0L;
+
+  // Track interval  every 30 seconds
+  private static final Duration WATCHER_INTERVAL = Duration.ofSeconds(30);
+
+  /** Watcher cluster lists */
+  private static final Map<Long, FlinkCluster> WATCHER_CLUSTERS = new ConcurrentHashMap<>(0);
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java]:
```suggestion
  private static final Map<Long, FlinkCluster> WATCHER_CLUSTERS = new ConcurrentHashMap<>(8);
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java [hunk 3]
```diff
@@ -160,16 +165,19 @@ public void start(FlinkCluster cluster) {
...
+        flinkCluster.setJobManagerUrl(deployResponse.address());
       } else {
         flinkCluster.setAddress(deployResponse.address());
       }
       flinkCluster.setClusterId(deployResponse.clusterId());
-      flinkCluster.setClusterState(ClusterState.STARTED.getValue());
+      flinkCluster.setClusterState(ClusterState.RUNNING.getValue());
       flinkCluster.setException(null);
       updateById(flinkCluster);
       FlinkRESTAPIWatcher.removeFlinkCluster(flinkCluster);
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java:177]: if the cluster is started successfully here, then we should add it into the monitor ?

In other scenerios with my limited read , `update`, `delete`,`shutdown` `start`, Its should be processed the logic by using FlinkClusterWatcher.addFlinkCluster or  removeFlinkCluster` if needed.

**Review by RocMarshal — COMMENTED** [review]
Hi,, @xujiangfeng001 
 Sorry for the late review & Thanks for your efforts on for items mentioned in the previous PR.

I left a few of comments,  Please let me know what's your opnion.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java (hunk 1 above)
Comment (xujiangfeng001) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java]: > ExecutionMode.isRemote(...

Thank you very much for the suggestion. It's a good suggestion

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java (hunk 2 above)
Comment (xujiangfeng001) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java]: > 

Thank you very much for the suggestion. It's a good suggestion

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java (hunk 3 above)
Comment (xujiangfeng001) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java:177]: > if the cluster is started successfully here, then we should add it into the monitor ?
> 
> In other scenerios with my limited read , `update`, `delete`,`shutdown` `start`, Its should be processed the logic by using FlinkClusterWatcher.addFlinkCluster or removeFlinkCluster` if needed.

This is my negligence. It is indeed necessary to remove it from the cache during `shutdown`, `start`.As for the `update`and `delete`scenarios, as the `flink cluster` is stopped, it means that there is no longer any `flink cluster` information in the cache, so I don't think it needs to be removed from the cache anymore.

**Comment by xujiangfeng001** [comment]
> Hi,, @xujiangfeng001 Sorry for the late review & Thanks for your efforts on for items mentioned in the previous PR.
> 
> I left a few of comments, Please let me know what's your opnion.

Hi @RocMarshal ，thank you very much for your suggestion. I have replied and made modifications to the relevant content. Looking forward to your next review

**Comment by RocMarshal** [comment]
thanks for the update.
LGTM +1. non-binding.

**Review by RocMarshal — DISMISSED** [review]

**Comment by xujiangfeng001** [comment]
> thanks for the update. LGTM +1. non-binding.

Thank you for your serious review.

**Review by wolfboys — COMMENTED** [review]
We need move the upgraded SQL to 2.2.0.sql file

**Comment by xujiangfeng001** [comment]
> We need move the upgraded SQL to 2.2.0.sql file

Thank you very much for your reminder. I have modified the upgraded SQL to 2.2.0.sql. Looking forward to your next review.

**Review by wolfboys — APPROVED** [review]
LGTM, Thanks for your hard work.

**Pull request merged**